src/runtime/posix_allocator.cpp

```diff
@@ -10,13 +10,16 @@ extern void free(void *);
 namespace Halide { namespace Runtime { namespace Internal {
 
 WEAK void *default_malloc(void *user_context, size_t x) {
-    void *orig = malloc(x+40);
+    // We want to return an aligned address to the application.
+    // In addition we should have at least the sizeof a pointer number
+    // of bytes so we can fit the original pointer for book keeping.
+    const size_t alignment = 128;
+    void *orig = malloc(x + alignment + sizeof(void *));
```

Comment:
x + alignment is already enough to store the original pointer. If we just wanted it to be aligned, then x + alignment - alignment_we_expect_from_malloc would be enough space. Note that the adjustment code below always adds alignment, even if the original pointer is already aligned, so there's always some space before the pointer. The +8 was to accomodate reading slightly off the end by one double, so this should be sizeof(double) not sizeof(void *).